src/ui-scroll.js

```diff
@@ -278,10 +281,10 @@ angular.module('ui.scroll', [])
 
         function insertWrapperContent(wrapper, insertAfter) {
           createElement(wrapper, insertAfter, viewport.insertElement);
-          if (!rowHeight && !isElementVisible(wrapper)) {
+          if (allowVisibilityWatch && !isElementVisible(wrapper)) {
             wrapper.unregisterVisibilityWatcher = wrapper.scope.$watch(() => visibilityWatcher(wrapper));
           }
-          if (!rowHeight) {
+          if (allowVisibilityWatch) {
```

Comment:
This particular case, I mean hiding the elements before data binding, is not related to visibility-watcher. I would say that it may relate to rowHeight. We are hiding new rows before data binding, because the height might change after binding, and there could be some bad side effects (scroll bar shifting). And we may not hide new rows if we know that the height is not going to be changed.